Sources_v3/Query/ChannelMemberListQuery.swift

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A query type used for fetching channel members from the backend.
+public struct ChannelMemberListQuery: Encodable {
+    private enum CodingKeys: String, CodingKey {
+        case filter = "filter_conditions"
+        case sort
+        case channelId = "id"
+        case channelType = "type"
+    }
+    
+    /// A channel identifier the members should be fetched for.
+    public let cid: ChannelId
+    /// A filter for the query (see `Filter`).
+    public let filter: Filter
+    /// A sorting for the query (see `Sorting`).
+    public let sort: [Sorting<ChannelMemberListSortingKey>]
+    /// A pagination.
+    public var pagination: Pagination
+    
+    /// Creates new `ChannelMemberListQuery` instance.
+    /// - Parameters:
+    ///   - cid: The channel identifier.
+    ///   - filter: The members filter.
+    ///   - sort: The sorting for members list.
+    ///   - pagination: The pagination.
+    public init(
+        cid: ChannelId,
+        filter: Filter,
+        sort: [Sorting<ChannelMemberListSortingKey>] = [],
+        pagination: Pagination = [.channelMembersPageSize]
+    ) {
+        self.cid = cid
+        self.filter = filter
+        self.sort = sort
+        self.pagination = pagination
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        
+        try container.encode(cid.id, forKey: .channelId)
+        try container.encode(cid.type, forKey: .channelType)
+        try container.encode(filter, forKey: .filter)
+        try pagination.encode(to: encoder)
+        if !sort.isEmpty { try container.encode(sort, forKey: .sort) }
+    }
+}
+
+extension ChannelMemberListQuery {
+    var hash: String {
+        filter.filterHash.appending(cid.rawValue)
```

Comment:
For `ChannelMemberListQuery` we cannot just go with **filter.filterHash** because we can have queries with the same filter but for different channels. So we should make the **cid** be a part of DTO identifier. 

Should we also use **sort** property when calculating the hash?

Reply:
Understood 👍 

I think `sort` should also be included because it affects pagination and it can cause issues.

Reply:
ok, will do 👍 The same applies for other queries, doesn't it?

Reply:
Yes 👍